lib/utils/typical/parser.go

```diff
@@ -491,6 +491,57 @@ func (e unaryFuncWithEnvExpr[TEnv, TArg, TResult]) Evaluate(env TEnv) (TResult,
 	return res, trace.Wrap(err, "evaluating function (%s)", e.name)
 }
 
+type unaryVariadicFunctionWithEnv[TEnv, TVarArgs, TResult any] struct {
+	impl func(TEnv, ...TVarArgs) (TResult, error)
+}
+
+// UnaryVariadicFunctionWithEnv returns a definition for a function that can be called
+// with any number of arguments with a single type. The [impl] will
+// be called with the evaluation env as the first argument, followed by the
+// actual arguments passed in the expression.
+func UnaryVariadicFunctionWithEnv[TEnv, TVarArgs, TResult any](impl func(TEnv, ...TVarArgs) (TResult, error)) Function {
```

Comment:
nit: I have all the other variadic function specs defined in a group below all the non-variadic ones, can this move down there after `UnaryVariadicFunction`? They are all very similar so if we need to change them, I think easier to change all at once that way

(One day I should probably define "WithEnv" for all function arities and make it the default/only option (and maybe use codegen here))